cou/steps/ceph.py

```diff
@@ -0,0 +1,104 @@
+# Copyright 2024 Canonical Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
```

Comment:
I thought that was the purpose of this module (see vault.py and nova_cloud_controller.py)

Reply:
Unfortunately all this is a lot of refactoring, and adds noise to the PR.  We could almost split this PR into 3 separate things now:

1. refactor plan warnings to support errors and non-blocking warnings
2. refactor ceph logic into its own module
3. add this set-noout feature.

Reply:
This is unfortunately has to be done in a single PR because they are dependent changes. If 1 and 2 are accepted and merged but the feature is not accepted, then the refactoring is useless. 

The refactoring should be done without any new features in mind, and we can focus on improving the "extendability" of COU iteself

Reply:
Thanks for updating the module. I agree that the changes increased for this PR, but beside some changes it's just moving around. I'm ok in resolving this thread if everybody agree

Reply:
Let's resolve the conversation and review these three topics together.